fastapi/dependencies/utils.py

```diff
@@ -462,10 +468,91 @@ async def solve_generator(
+    stack: Union[None, AsyncExitStack]
+) -> Tuple[
+    Dict[str, Any],
+    Dict[Tuple[Callable[..., Any], Tuple[str]], Any],
+]:
+    values: Dict[str, Any] = {}
+    sub_dependant: Dependant
+    dependency_cache = dependency_cache or app_dependency_cache.copy()
+    for sub_dependant in dependant.dependencies:
+        sub_dependant.call = cast(Callable[..., Any], sub_dependant.call)
+        sub_dependant.cache_key = cast(
+            Tuple[Callable[..., Any], Tuple[str]], sub_dependant.cache_key
+        )
+        call = sub_dependant.call
+        use_sub_dependant = sub_dependant
+        if (
+            dependency_overrides_provider
+            and dependency_overrides_provider.dependency_overrides
+        ):
+            original_call = sub_dependant.call
+            call = getattr(
+                dependency_overrides_provider, "dependency_overrides", {}
+            ).get(original_call, original_call)
+            use_sub_dependant = get_dependant(
+                call=call,
+                name=sub_dependant.name,
+            )
+        solved_result = await solve_lifespan_dependencies(
+            dependant=use_sub_dependant,
+            dependency_overrides_provider=dependency_overrides_provider,
+            app_dependency_cache=app_dependency_cache,
+            dependency_cache=dependency_cache,
+            stack=stack
+        )
+        sub_values, sub_dependency_cache = solved_result
+        dependency_cache.update(sub_dependency_cache)
+        if sub_dependant.use_cache != DependencyCacheScope.nocache and sub_dependant.cache_key in dependency_cache:
+            solved = dependency_cache[sub_dependant.cache_key]
+        elif is_gen_callable(call) or is_async_gen_callable(call):
+            if stack is None:
+                raise RuntimeError(
+                    async_contextmanager_dependencies_error
+                )  # pragma: no cover
+            solved = await solve_generator(
+                call=call, stack=stack, sub_values=sub_values
+            )
+        elif is_coroutine_callable(call):
+            solved = await call(**sub_values)
+        else:
+            solved = await run_in_threadpool(call, **sub_values)
+        if sub_dependant.name is not None:
+            values[sub_dependant.name] = solved
+        if sub_dependant.cache_key not in dependency_cache:
+            dependency_cache[sub_dependant.cache_key] = solved
+        if sub_dependant.use_cache == DependencyCacheScope.app and sub_dependant.cache_key not in app_dependency_cache:
+            app_dependency_cache[sub_dependant.cache_key] = solved
+    if any((
+        dependant.request_param_name,
+        dependant.query_params,
+        dependant.header_params,
+        dependant.cookie_params,
+        dependant.body_params,
+        dependant.request_param_name,
+        dependant.websocket_param_name,
+        dependant.http_connection_param_name,
+        dependant.response_param_name,
+        dependant.path
```

Comment:
I hope this is a sensible set of exclusions. I dunno what to do with the security dependencies.